src/server/api/general.js

```diff
@@ -124,32 +123,11 @@ export default function useGeneralApi(app) {
         }
 
         // acquire global lock so only one account can be created at a time
-        try {
-            const lock_entity_res = yield Tarantool.instance().call(
-                'lock_entity',
-                user_id + ''
-            );
-            if (!lock_entity_res[0][0]) {
-                console.log(
-                    '-- /accounts lock_entity -->',
-                    user_id,
-                    lock_entity_res[0][0]
-                );
-                this.body = JSON.stringify({ error: 'Conflict' });
-                this.status = 409;
-                return;
-            }
-        } catch (e) {
-            console.error(
-                '-- /accounts tarantool is not available, fallback to another method',
-                e
-            );
-            const rnd_wait_time = Math.random() * 10000;
-            console.log('-- /accounts rnd_wait_time -->', rnd_wait_time);
-            yield new Promise(resolve =>
-                setTimeout(() => resolve(), rnd_wait_time)
-            );
-        }
+        const rnd_wait_time = Math.random() * 10000;
```

Comment:
wat

Reply:
Seriously, what? is this supposed to be a spinlock? i get that this isn't your code, but exactly what is happening here?

Reply:
This is not even remotely idiomatic Javascript. Can you please repair this?

Reply:
+1 idk why it has to be random. this is the status quo, though.

Reply:
just discussed this on the call; the random wait can be removed.

Reply:
@goldibex, I've added this as fallback solution for those rare cases when Tarantool db is down/not available. I admit its pure quality from idea to implementation - I was time constrained at this moment and the issue I was trying to fix was a real exploit costing us a couple of hundred bot accounts registered each day.